### PR TITLE
Updating mocks for Orders, Products and Reviews screens as preparation for UI tests

### PR DESCRIPTION
**Pull request by jostnes**
### Description
This PR add/update existing mocks so that content are displayed on Orders, Products and Reviews page. Added `testGotoOrdersScreen()` test which I used to test that the mocked content were displayed. Work needs to be done on that to make it a proper test (which will be in a separate PR).

🙏  for the number of files changed, most of those are mocked data/json files being added/updated.

### Testing instructions
Made sure existing UI tests are still passing

### Screenshots
Orders screen mocked content displayed:

Screenshot 2021-12-02 at 10 53 14 AM

Products screen:

Screenshot 2021-12-02 at 10 53 23 AM

Reviews screen:

Screenshot 2021-12-02 at 10 53 35 AM
---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/UITestsFoundation/Screens/Orders/OrdersScreen.swift
Comment (pachlava) [inline WooCommerce/UITestsFoundation/Screens/Orders/OrdersScreen.swift]: The `SceenObject` class, which `OrdersScreen` instantiates, already has a _public_ `isLoaded` method. That method will check for the existence of the first element from `expectedElementGetters`.  So, current method is not necessary:

```suggestion
```

File: WooCommerce/UITestsFoundation/Screens/Orders/OrdersScreen.swift
Comment (pachlava) [inline WooCommerce/UITestsFoundation/Screens/Orders/OrdersScreen.swift:29]: Once the local `isLoaded` is removed, and the `testGotoOrdersScreen` will now refer to `isLoaded` from `ScreenObject`, technically there's no need to keep this public:

```suggestion
   init(app: XCUIApplication = XCUIApplication()) throws {
```

However, keeping it `public` is not a crime at the same time, IMO, just sharing my observations 🙂 Feel free to keep it the way you prefer.

File: WooCommerce/UITestsFoundation/Screens/TabNavComponent.swift
Comment (pachlava) [inline WooCommerce/UITestsFoundation/Screens/TabNavComponent.swift:27]: Because of reasons mentioned in neigboring comments, this can be reverted:

```suggestion
   init(app: XCUIApplication = XCUIApplication()) throws {
```

File: WooCommerce/UITestsFoundation/Screens/TabNavComponent.swift
Comment (pachlava) [inline WooCommerce/UITestsFoundation/Screens/TabNavComponent.swift]: Because of reasons mentioned in nearby comments, this can be reverted:

```suggestion
        if !OrdersScreen.isVisible {
```

`isVisible` actually calls the `isLoaded` from `ScreenObject`

File: WooCommerce/WooCommerceUITests/Tests/OrdersTests.swift
Comment (pachlava) [inline WooCommerce/WooCommerceUITests/Tests/OrdersTests.swift]: (Keeping in mind that this is a temporary and unfinished test) Generally, the login is a part of `setUp`. In cases if it's not, it's natural to include the logout into the `tearDown`, to make sure the test leaves the app in the state before the test.

File: WooCommerce/WooCommerceUITests/Tests/OrdersTests.swift
Comment (pachlava) [inline WooCommerce/WooCommerceUITests/Tests/OrdersTests.swift]: Just a line spacing nitpick, for consistency:

```suggestion
final class OrdersTests: XCTestCase {

    override func setUp() {
```

File: WooCommerce/WooCommerceUITests/Flows/LoginFlow.swift
Comment (pachlava) [inline WooCommerce/WooCommerceUITests/Flows/LoginFlow.swift]: Good job on extracting this into a separate flow class 👍 

One nitpick: WDYT about renaming it to something like this:

```suggestion
    static func logInWithWPcom() throws -> MyStoreScreen {
```

Which will make a comment not necessary then (also me being extra-nitpicky about logIn being a verb in this case, not a noun).

**Review by pachlava — CHANGES_REQUESTED** [review]
Thanks for this noticeable amount of foundation work @jostnes !

As usual, I started from all tests execution on iPhone 13 with iOS 15, and all passed. Meaning, there are no regressions. Besides this, I just initiated the run on CI (it's optional and needs approval) - we'll see how it goes there.

Also, I navigated the app manually, and could see the data loading properly (images too) in Products, Orders and Reviews 👍.

As we agreed yesterday, I tried to be nitpicky 🙂, and left some notes. Only one of them (removal of the duplicated `isLoaded`) I consider necessary to address before merging.

I did not review the mocks files, only the swift ones - there might be need to review jsons in future, once we are in the context of a specific app area / test.

**Comment by pachlava** [comment]
All UI tests (including the new one) [passed](https://app.circleci.com/pipelines/github/woocommerce/woocommerce-ios/15886/workflows/6ae877c3-20b5-48ef-820c-c8f4a0bbaebb) on CI.

File: WooCommerce/UITestsFoundation/Screens/Orders/OrdersScreen.swift
Comment (jostnes) [reply WooCommerce/UITestsFoundation/Screens/Orders/OrdersScreen.swift:29]: I was getting this error `'OrdersScreen' initializer is inaccessible due to 'internal' protection level` when building the app earlier, adding `public` seems to have fixed it. Will keep this here for now since it is harmless.

File: WooCommerce/UITestsFoundation/Screens/Orders/OrdersScreen.swift
Comment (jostnes) [reply WooCommerce/UITestsFoundation/Screens/Orders/OrdersScreen.swift]: Ah ok, I've removed this (and all the references) and it does work. This has been reverted in the latest commit.

File: WooCommerce/UITestsFoundation/Screens/Orders/OrdersScreen.swift
Comment (pachlava) [reply WooCommerce/UITestsFoundation/Screens/Orders/OrdersScreen.swift:29]: Yes, this is because you were using the local `isLoaded` previously. But as you say, it's not harmful to keep it this was 👍

File: WooCommerce/WooCommerceUITests/Flows/LoginFlow.swift
Comment (pachlava) [inline WooCommerce/WooCommerceUITests/Flows/LoginFlow.swift:6]: Since the function name was made more descriptive, as a step to self-documenting code, I think the comment is not necessary anymore:

```suggestion
```

**Review by pachlava — APPROVED** [review]
Thanks for the adjustments @jostnes ! Left a few observational notes, the CI is green, so it's good to go! :shipit:

File: WooCommerce/WooCommerceUITests/Flows/LoginFlow.swift
Comment (jostnes) [reply WooCommerce/WooCommerceUITests/Flows/LoginFlow.swift:6]: Makes sense, I'll remove this in the next PR.

**Comment by jostnes** [comment]
> Thanks for the adjustments @jostnes ! Left a few observational notes, the CI is green, so it's good to go! :shipit:

Thanks @pachlava for the thorough review!

**Pull request merged**
